src/test/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/security/ElasticsearchCaseSearchRequestSecurityTest.java

```diff
@@ -73,4 +78,30 @@ void shouldSecureRequest() {
             () -> verify(objectMapperService, times(2)).convertStringToObject(anyString(), eq(ObjectNode.class))
         );
     }
+
+    @Test
+    @DisplayName("should parse and secure request with filters and multiple case types")
+    void shouldSecureCrossCaseTypeRequest() {
+        List<String> caseTypeIds = new ArrayList<>();
+        caseTypeIds.add(CASE_TYPE_ID);
+        caseTypeIds.add(CASE_TYPE_ID_2);
+
+        CrossCaseTypeSearchRequest request = mock(CrossCaseTypeSearchRequest.class);
+        doReturn(caseTypeIds).when(request).getCaseTypeIds();
+        doReturn(new ElasticsearchRequest(searchRequestJsonNode)).when(request).getElasticSearchRequest();
+        doReturn(true).when(request).isMultiCaseTypeSearch();
+        doReturn(Optional.of(mock(QueryBuilder.class))).when(caseSearchFilter).getFilter(CASE_TYPE_ID);
+        doReturn(searchRequestJsonNode).when(objectMapperService).convertStringToObject(anyString(),
+            eq(ObjectNode.class));
+        doReturn(searchRequestJsonNode).when(searchRequestJsonNode).get(anyString());
+
+        querySecurity.createSecuredSearchRequest(request);
+
+        assertAll(
+            () -> verify(caseSearchFilter).getFilter(CASE_TYPE_ID),
+            () -> verify(request).isMultiCaseTypeSearch(),
+            () -> verify(objectMapperService, times(1)).convertStringToObject(anyString(), eq(ObjectNode.class))
```

Comment:
instead of `anyString()` can we verify  actual query string constructed. Use proper Object builders if required.

Reply:
Agreed: it would be nice if this was expanded to prove output.